saltstack/base/salt/luks/bin/create-luks-volume.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+set -ex -o pipefail
+
+: ${SECRET_ENCRYPTION_KEY_SOURCE:? required}
+
+export LUKS_BACKING_FILE_DEFAULT_SIZE="100MiB"
+
+if mountpoint /mnt/luks; then
+  echo "Volume already mounted at mount point... Exiting LUKS volume creation!"
+  exit
+fi
+
+if [[ -e /etc/luks/luks && $(stat -c "%a" /etc/luks/luks | tr -d '\n') == 600 ]]; then
+  # Set the backing file to the specified size if it exists
+  truncate -s "$LUKS_BACKING_FILE_DEFAULT_SIZE" /etc/luks/luks
+else
+  # Create the backing file if it doesn't exist
+  dd if=/dev/zero of=/etc/luks/luks bs="$LUKS_BACKING_FILE_DEFAULT_SIZE" count=1
+  chmod 600 /etc/luks/luks
+fi
+
+if ! mountpoint /mnt/luks_passphrase_tmpfs; then
+  # Create the tmpfs for the plaintext passphrase
+  mount -t tmpfs -o size=1k,mode=700 tmpfs /mnt/luks_passphrase_tmpfs
+fi
+
+if [[ -e /mnt/luks_passphrase_tmpfs/passphrase ]]; then
+  echo "Plaintext passphrase already exists... Exiting LUKS volume creation with failed exit code!"
+  exit 1
+else
+  aws kms generate-random \
+      --number-of-bytes 64 \
+      --output text \
+      --query Plaintext \
+    | base64 --decode \
+    1> /mnt/luks_passphrase_tmpfs/passphrase
```

Comment:
Nit: Out of curiosity, why specify the file descriptor explicitly? `>` by default operates on stdout / file descriptor 1.